tests/integration/test_types_integration.py

```diff
@@ -142,6 +145,166 @@ def test_json(trino_connection):
         .execute()
 
 
+def test_date(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS DATE)", python=None) \
+        .add_field(sql="DATE '2001-08-22'", python=date(2001, 8, 22)) \
+        .add_field(sql="DATE '0001-01-01'", python=date(1, 1, 1)) \
+        .add_field(sql="DATE '1582-10-04'", python=date(1582, 10, 4)) \
+        .add_field(sql="DATE '1582-10-05'", python=date(1582, 10, 5)) \
+        .add_field(sql="DATE '1582-10-14'", python=date(1582, 10, 14)) \
+        .execute()
```

Comment:
You can't combine comments and `\`, so this wouldn't actually work in Python. 

IMHO this setup is not really great in the sense that you never know which value errors out. Probably we should switch to parametrised tests.

Reply:
The alternative is to catch the failure and then rerun the cases one by one instead of a single query like we do right now. Parametrization over so many statements will be slow.

Reply:
> You can't combine comments and \, so this wouldn't actually work in Python.

I know hence the `()` outside and no `\`.

Reply:
Oh oops I missed that.